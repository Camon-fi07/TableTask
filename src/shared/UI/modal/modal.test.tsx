import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Modal, ModalContent, ModalHeader, ModalHeaderTitle } from './';

const Component = ({ onClose }: { onClose: () => void }) => (
  <Modal data-testid='modal' onClose={onClose}>
    <ModalHeader>
      <ModalHeaderTitle>Заголовок</ModalHeaderTitle>
    </ModalHeader>
    <ModalContent>
      <div>text</div>
      <div>text</div>
      <div>text</div>
    </ModalContent>
  </Modal>
);

describe('modal', () => {
  const onClose = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render', () => {
    const { getByTestId } = render(<Component onClose={onClose} />);
    expect(getByTestId('modal')).toBeInTheDocument();
  });

  it('should call onClose when clicking on close button', () => {
    const { getByTestId } = render(<Component onClose={onClose} />);
    const closeModalButton = getByTestId('close-modal-button');
    userEvent.click(closeModalButton);
    expect(onClose).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { getByTestId } = render(<Component onClose={onClose} />);
    expect(getByTestId('modal')).toMatchSnapshot();
  });
});
