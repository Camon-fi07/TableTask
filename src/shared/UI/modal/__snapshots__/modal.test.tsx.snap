// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal should match snapshot 1`] = `
<div
  class="modal"
  data-testid="modal"
>
  <div
    class="modalHeader"
  >
    <h3
      class="modalHeaderTitle"
    >
      Заголовок
    </h3>
  </div>
  <div
    class="modalContent"
  >
    <div>
      text
    </div>
    <div>
      text
    </div>
    <div>
      text
    </div>
  </div>
  <button
    class="closeButton"
    data-testid="close-modal-button"
  >
    <div
      class="closeButtonContent"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-x"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18 6 6 18"
        />
        <path
          d="m6 6 12 12"
        />
      </svg>
    </div>
  </button>
</div>
`;
